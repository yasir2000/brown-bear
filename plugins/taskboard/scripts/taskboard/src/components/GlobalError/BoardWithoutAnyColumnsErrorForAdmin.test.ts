/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { Vue } from "vue/types/vue";
import { shallowMount } from "@vue/test-utils";
import { createTaskboardLocalVue } from "../../helpers/local-vue-for-test";
import BoardWithoutAnyColumnsErrorForAdmin from "./BoardWithoutAnyColumnsErrorForAdmin.vue";
import { createStoreMock } from "../../../../../../../src/scripts/vue-components/store-wrapper-jest";

describe("BoardWithoutAnyColumnsError", () => {
    let local_vue: typeof Vue;

    beforeEach(async () => {
        local_vue = await createTaskboardLocalVue();
    });

    it("is displays misconfiguration error for admin user", () => {
        const wrapper = shallowMount(BoardWithoutAnyColumnsErrorForAdmin, {
            localVue: local_vue,
            mocks: { $store: createStoreMock({ state: { admin_url: "/path/to/admin" } }) },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
