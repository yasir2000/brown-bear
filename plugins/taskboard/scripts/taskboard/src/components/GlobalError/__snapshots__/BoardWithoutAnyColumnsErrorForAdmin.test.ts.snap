// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardWithoutAnyColumnsError is displays misconfiguration error for admin user 1`] = `
<section
  class="empty-state-page"
>
  <div
    class="empty-state-illustration"
  >
    <!---->
  </div>
   
  <h1
    class="empty-state-title"
  >
    <translate-stub
      tag="span"
    >
      This taskboard is not properly configured
    </translate-stub>
  </h1>
   
  <p
    class="empty-state-text"
  >
    <translate-stub
      tag="span"
    >
      You have to configure the columns mapping.
    </translate-stub>
  </p>
   
  <a
    class="tlp-button-primary empty-state-action"
    href="/path/to/admin"
  >
    <i
      aria-hidden="true"
      class="fas fa-long-arrow-alt-right tlp-button-icon"
    />
     
    <translate-stub
      tag="span"
    >
      Define columns
    </translate-stub>
  </a>
</section>
`;
